arch/arm/src/stm32/hardware/stm32g47xxx_rcc.h

```diff
@@ -0,0 +1,677 @@
+#  define RCC_PLLCFGR_PLLSRC_HSE       (3 << RCC_PLLCFGR_PLLSRC_SHIFT)
+                                                           /* Bits 3-2: Reserved */
+#define RCC_PLLCFGR_PLLM_SHIFT         (4)                 /* Bits 7-4: Division factor M of main PLL input clock */
+#define RCC_PLLCFGR_PLLM_MASK          (0xf << RCC_PLLCFGR_PLLM_SHIFT)
+#define RCC_PLLCFGR_PLLM(n)            ((((n) - 1) << RCC_PLLCFGR_PLLM_SHIFT) & RCC_PLLCFGR_PLLM_MASK) /* n=1..16 */
+
+#define RCC_PLLCFGR_PLLN_SHIFT         (8)                 /* Bits 14-8: Main PLL multiplication factor N for VCO */
+#define RCC_PLLCFGR_PLLN_MASK          (0x7f << RCC_PLLCFGR_PLLN_SHIFT)
+#define RCC_PLLCFGR_PLLN(n)            (((n) << RCC_PLLCFGR_PLLN_SHIFT) & RCC_PLLCFGR_PLLN_MASK) /* n=8..127 */
+                                                           /* Bit 15: Reserved */
+#define RCC_PLLCFGR_PLLPEN             (1 << 16)           /* Bit 16: Main PLL PCLK output enable */
+#define RCC_PLLCFGR_PLLP               (1 << 17)           /* Bit 17: Main PLL division factor for PLL P clock, 0:PLLP=7, 1:PLLP=17 */
+                                                           /* Bits 19-18: Reserved */
+#define RCC_PLLCFGR_PLLQEN             (1 << 20)           /* Bit 20: Main PLL Q clock output enable */
+#define RCC_PLLCFGR_PLLQ_SHIFT         (21)                /* Bits 22-21: Main PLL division factor for PLL Q clock */
+#define RCC_PLLCFGR_PLLQ_MASK          (0x3 << RCC_PLLCFGR_PLLQ_SHIFT)
+#  define RCC_PLLCFGR_PLLQ_2           (0x0 << RCC_PLLCFGR_PLLQ_SHIFT)
+#  define RCC_PLLCFGR_PLLQ_4           (0x1 << RCC_PLLCFGR_PLLQ_SHIFT)
+#  define RCC_PLLCFGR_PLLQ_6           (0x2 << RCC_PLLCFGR_PLLQ_SHIFT)
+#  define RCC_PLLCFGR_PLLQ_8           (0x3 << RCC_PLLCFGR_PLLQ_SHIFT)
+                                                           /* Bits 23: Reserved */
+#define RCC_PLLCFGR_PLLREN             (1 << 24)           /* Bit 24: PLL R clock output enable */
+#define RCC_PLLCFGR_PLLR_SHIFT         (25)                /* Bits 26-25: Main PLL division factor for PLL R clock (system clock) */
+#define RCC_PLLCFGR_PLLR_MASK          (3 << RCC_PLLCFGR_PLLR_SHIFT)
+#  define RCC_PLLCFGR_PLLR_2           (0x0 << RCC_PLLCFGR_PLLR_SHIFT)
+#  define RCC_PLLCFGR_PLLR_4           (0x1 << RCC_PLLCFGR_PLLR_SHIFT)
+#  define RCC_PLLCFGR_PLLR_6           (0x2 << RCC_PLLCFGR_PLLR_SHIFT)
+#  define RCC_PLLCFGR_PLLR_8           (0x3 << RCC_PLLCFGR_PLLR_SHIFT)
+
+#define RCC_PLLCFGR_PLLPDIV_SHIFT      (27)                /* Bits 31-27: Main PLL division factor for PLL P clock */
+#define RCC_PLLCFGR_PLLPDIV_MASK       (0x1f << RCC_PLLCFGR_PLLPDIV_SHIFT)
+#define RCC_PLLCFGR_PLLPDIV(n)         (((n) << RCC_PLLCFGR_PLLPDIV_SHIFT) & RCC_PLLCFGR_PLLPDIV_MASK) /* n=2..31 */
+
+#define RCC_PLLCFGR_RESERVED_MASK      0x008c800c          /* Bits 23, 19-18, 15, and 3-2 should be kept at reset value */
+#define RCC_PLLCFGR_RESET              0x00001000
+
+/* CIER - Clock Interrupt Enable Register */
+
```

Comment:
I try to do this files as a representation of the reference manual at the version I am working off of. It is a PITA but keeping the names and the comments consistent with the manual help to speed development and catch the bugs as the docs evolve. "Oh this is what we thought it does, but now the new rev of the manual says it does this"

Reply:
Agreed, and I tried to do the same wherever possible. What isn't consistent with the manual here?